src/Event/Productions/EventCannotBeAddedToProduction.php

```diff
@@ -6,17 +6,24 @@
 
 final class EventCannotBeAddedToProduction extends Exception
 {
-    public static function becauseItAlreadyBelongsToAnotherProduction(string $eventId, ProductionId $productionId)
+    public static function becauseItAlreadyBelongsToAnotherProduction(string $eventId, ProductionId $productionId): self
     {
         return new self(
             'Event with id ' . $eventId . ' cannot be added to production with id ' . $productionId->toNative() . ' because it already belongs to another production.'
         );
     }
 
-    public static function becauseSomeEventsBelongToAnotherProduction(array $eventIds, ProductionId $productionId)
+    public static function becauseSomeEventsBelongToAnotherProduction(array $eventIds, ProductionId $productionId): self
     {
         return new self(
             'Events with ids ' . join(',', $eventIds) . ' cannot be added to production with id ' . $productionId->toNative() . ' because some events already belong to another production.'
```

Comment:
I get a warning in PHPStorm that `join` is an alias for `implode` and it is preferred to used `implode`.